middleware/http/opa/middleware.go

```diff
@@ -19,23 +19,30 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"math"
+	"net/http"
+	"net/textproto"
 	"strconv"
 	"strings"
 
 	"github.com/open-policy-agent/opa/rego"
-	"github.com/valyala/fasthttp"
+	"k8s.io/utils/strings/slices"
 
+	"github.com/dapr/components-contrib/internal/httputils"
+	"github.com/dapr/components-contrib/internal/utils"
 	"github.com/dapr/components-contrib/middleware"
 	"github.com/dapr/kit/logger"
 )
 
 type Status int
 
 type middlewareMetadata struct {
-	Rego            string `json:"rego"`
-	DefaultStatus   Status `json:"defaultStatus,omitempty"`
-	IncludedHeaders string `json:"includedHeaders,omitempty"`
+	Rego                  string   `json:"rego"`
+	DefaultStatus         Status   `json:"defaultStatus,omitempty"`
+	IncludedHeaders       string   `json:"includedHeaders,omitempty"`
+	SkipBody              string   `json:"skipBody,omitempty"`
```

Comment:
Thinking about it, given how this is an alpha component, I wonder if it would be best to change the behavior to NOT read the body by default.

Essentially, the body would not be read (for performance reasons - to avoid reading it all in memory), unless `readBody` is set in the metadata.

This is a breaking change for people who depend on the body to create OPA policies. However, it seems to be a saner default.

@berndverst thoughts?

Reply:
Sure, sounds good -- add the new metadata option `readBody` which is a boolean with default of `false`